### PR TITLE
remove include

### PR DESCRIPTION
**Pull request by pavel-pimenov**

### CODE REVIEW

File: examples/client_test.cpp [hunk 1]
```diff
@@ -30,7 +30,6 @@ POSSIBILITY OF SUCH DAMAGE.
 
 */
 
-#include <cstdio> // for snprintf
```
Comment (arvidn) [inline examples/client_test.cpp]: `std::snprintf()` is used in this file. It seems inappropriate to remove the include and instead rely on some other header including this "by accident".

**Comment by arvidn** [comment]
how did you determine which includes should be removed? It doesn't look like the algorithm is all the way right.

**Comment by pavel-pimenov** [comment]
Doxygen include graph for client_test.cpp
...
#include "print.hpp" 
...

изображение

**Comment by arvidn** [comment]
I don't think it's good practice to rely on other headers to pull in dependencies the file itself has. it makes the headers and cpp files less independent. includes that don't have any direct dependecies in the file though, those are good candidates for removal.

File: test/web_seed_suite.cpp [hunk 2]
```diff
@@ -30,14 +30,12 @@ POSSIBILITY OF SUCH DAMAGE.
 
 */
 
-#include "libtorrent/session.hpp"
```
Comment (arvidn) [inline test/web_seed_suite.cpp]: this file uses `session`. it constructs one. I don't see why you wouldn't want to include the header

Comment (pavel-pimenov) [reply test/web_seed_suite.cpp]: изображение

Comment (arvidn) [reply test/web_seed_suite.cpp]: I don't understand your point. `setup_transfer.hpp` *also* uses `session`, right? so it makes sense for it to also include it.
To put it another way, if `setup_transfer.hpp` for some reason, in the future, stops using `session` and removes its include; it doesn't make any sense for `web_seed_suite.cpp` to fail.

**Comment by arvidn** [comment]
@pavel-pimenov I'm not really interested in manually reviewing all includes you've removed, to make sure they are sensible. Instead, can you tell me what algorithm you've used to determine what to remove?

**Pull request closed without merging**